core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -486,10 +487,10 @@ protected final GetResult getFromSearcher(Get get, Function<String, Searcher> se
     }
 
     public final GetResult get(Get get) throws EngineException {
-        return get(get, this::acquireSearcher);
+        return get(get, this::acquireSearcher, null);
     }
 
-    public abstract GetResult get(Get get, Function<String, Searcher> searcherFactory) throws EngineException;
+    public abstract GetResult get(Get get, Function<String, Searcher> searcherFactory, @Nullable MeanMetric refreshMetric) throws EngineException;
```

Comment:
Sorry, I do not think this should be done by pushing the metric into the engine. Now there are two places the metric can be updated and it's mixing things up. This needs to stay at the index shard level and can be done via callbacks.